src/pages/detail/detail.tsx

```diff
@@ -1,21 +1,45 @@
+import React, { useEffect, useState } from "react";
 import { useParams } from "react-router-dom";
+import { useTokenContext } from "../../context";
+import { DefaultInfo } from "../../interface";
+import { getOneTask } from "../../utils/";
+
+import Spinner from 'react-bootstrap/Spinner';
 
 export interface ParamsType {
     id: string;
 }
 
 const Detail: React.FC = () => {
-
+    const {token} = useTokenContext();
     let {id} = useParams<ParamsType>();
-    const data = JSON.parse(localStorage.getItem(id) ?? '');
+    const [data, setData] = useState(DefaultInfo);
+    const [loading, setLoading] = useState(true);
+
+    useEffect(()=>{
+        getOneTask(token, id)
+        .then(res => {
+            if(res.status){
+                setData(res.data)
+                setLoading(false);
+            }
+        })
+    }, [token, id])
 
     return(
-        <div style={{textAlign: 'left', padding: '5% 10%'}}>
-            <p><strong>Description: </strong> {data.description}</p>
-            <p><strong>Category: </strong> {data.Category}</p>
-            <p><strong>content: </strong> {data.content}</p>
-            <p><strong>deadline: </strong> {data.deadline}</p>
-        </div>
+    <React.Fragment>
+        {loading? (
+            <Spinner animation="grow" style={{marginTop: '5rem'}}/>

```

Comment:
This `<Spinner> ..` shows up a few times in your PR, consider extracting it into a shared component.